### PR TITLE
Adds cages

### PR DESCRIPTION
**Pull request by unid15**
- Cages take 6 plasteel sheets and 10 seconds to build.
- Cages have double-layered doors. Closing the inner layer prevents mobs inside from moving, however they can still interact with their surroundings (and be interacted with). Closing the outer layer prevents mobs from doing anything but yelling "HELP" (and attempting to break out)
- Cages can be used to safely store and transport dangerous animals. They'll never break out of them!
- Humans can break out of the cage in 210 seconds. Opening the outer layer takes 30 seconds, opening inner takes 180 seconds. Having the HULK and STRONG mutations can reduce the required time to 65 seconds (5 + 60)
- It takes 3 seconds to close a cage.

### CODE REVIEW

**Comment by ghost** [comment]
Make it something between 2 and 5 plasteel, not 30

Roboticists would kill for a single sheet wasted, they won't like it if the Clown pilfers the E.V.A's reserve just to build one cage

**Comment by Imagebard** [comment]
How the fuck do we get mobs stunned so we can enter tough?
The safest way to do that is put a full suit of riot armor and trying to close the cage before the monster damages you too much

**Comment by Imagebard** [comment]
Also is there any interaction when trying to use the "jump" geneticist power inside the cage?

**Comment by FudgePucker** [comment]
Yea three seconds is too long for mobs, but too short to prevent disarm shove.

**Comment by ghost** [comment]
@Imagebard Would be neat if jumping in a certain direction while in the box makes said box move in that direction.

While also giving brain damage if you don't have proper headgear on

**Comment by ghost** [comment]
Also yeah, if the close time's gonna be three seconds, you're gonna need something to stun or at least hold a creature in place or in a controllable manner.

Lasso's when?

**Comment by ihadtoregisterforthis** [comment]
Netguns when?

**Comment by ghost** [comment]
What's with the silly delays? 3 seconds to close is too long.

20 seconds to craft is ridiculously long. Probably the longest action aside from breaking from cuffs/lockers.

File: code/__HELPERS/_macros.dm [hunk 1]
```diff
@@ -131,6 +131,10 @@
 
 #define isdeathsquad(H) (H.mind in ticker.mode.deathsquads)
 
+
+#define ishulk(H) (M_HULK in H.mutations)
```
Comment (9600bauds) [inline code/__HELPERS/_macros.dm]: I'm pretty sure this is fucked up and doesn't actually do what you think it does, see #7092

**Comment by unid15** [comment]
@TermedClepe since breaking out of a cage is a huge pain in the ass, the delays are here to make it harder to cage random people in the hallways. Construction delay prevents you from building a cage on the spot, so you have to drag one behind you. Opening/closing delay is so that you require a stun to capture somebody

**Comment by ghost** [comment]
Instant welderlocking in hallways isn't happening and is completely possible already, though I guess merely closing a cage on someone would be easier.

10 seconds is much better.

**Comment by ghost** [comment]
When you break out of a cage, does it become broken similarly to a locker? As in it can't be locked again.

Can the cages be wrenched to the floor? Perhaps welded afterwards for a more permanent display, like emitters.

**Comment by unid15** [comment]
Nope, when you resist out of a cage it only opens

They can be wrenched to the floor, yeah

**Comment by ghost** [comment]
Awesome, can't wait to make cages for greyshirts in brig.

**Comment by unid15** [comment]
ready for action

**Comment by gbasood** [comment]
Prepare for exploits.

File: code/game/objects/structures/cage.dm [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	if((damage >= damage_threshold_to_break))
+		if(prob(80))
+			visible_message("<span class='warning'>\The [src] shakes violently!</span>")
+		else
+			toggle_door(user)
+
+/obj/structure/cage/examine(mob/user)
+	..()
+
+	to_chat(user, "<span class='info'>Alt + click opens/closes the cage's cover.</span>")
```
Comment (ihadtoregisterforthis) [inline code/game/objects/structures/cage.dm:53]: Thankfully this is a low-RP server.

Comment (ghost) [inline code/game/objects/structures/cage.dm:53]: No no no no NO MUH IMMERSIONS

Comment (unid15) [inline code/game/objects/structures/cage.dm:53]: how else do you let people know how to use a new feature (considering nobody reads the changelog)

Comment (ghost) [inline code/game/objects/structures/cage.dm:53]: Remove this

File: code/game/objects/structures/cage.dm [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	if(iswrench(W))
+		if(anchored)
+			to_chat(user, "<span class='info'>You start unsecuring \the [src] from \the [loc].</span>")
+		else
+			if(!istype(loc, /turf/simulated/floor)) //Can't secure the cage to space
+				return
+
+			to_chat(user, "<span class='info'>You start securing \the [src] to \the [loc].</span>")
+
+		spawn()
```
Comment (ComicIronic) [inline code/game/objects/structures/cage.dm:69]: Why would you do such a thing

File: code/game/objects/structures/cage.dm (hunk 2 above)
Comment (ComicIronic) [inline code/game/objects/structures/cage.dm:53]: Why would you DO such a thing

Comment (PJB3005) [inline code/game/objects/structures/cage.dm:53]: This was removed by the way.

**Pull request merged**